Python/ceval.c

```diff
@@ -3135,8 +3135,11 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             PyObject *getattribute = read_obj(cache->descr);
             assert(Py_IS_TYPE(getattribute, &PyFunction_Type));
             PyFunctionObject *f = (PyFunctionObject *)getattribute;
+            uint32_t func_version = read_u32(cache->keys_version);
+            assert(func_version != 0);
+            DEOPT_IF(f->func_version != func_version, LOAD_ATTR);
```

Comment:
Is it possible to add a test for this deoptimisation?

Reply:
is there an existing test I can use as a reference for how to trigger this specialization? I couldn't find any from a cursory search

Reply:
This specialisation triggers with the following code:

```python

class X:
    def __gettattribute__(self, attr):
        return attr

x = X()
for _ in range(8):
    x.a  #LOAD_ATTR_GETATTRIBUTE_OVERRIDDEN
```

Reply:
thanks, I tried devising a test for this, PTAL

Reply:
Thanks! I gave it a try, but unfortunately, the test does not fail if I remove the deopt.

Reply:
thanks for checking @erlend-aasland ! 

I realized I was testing something else (that specialization doesn't trigger for an already-overridden vectorcall)

pushed an update that now tests both scenarios (avoid specialization, and deopt) - verified the `test_setvectorcall_load_attr_specialization_deopt` fails when removing the deopt

Reply:
Thanks, looks good! :)